### PR TITLE
DNM: Resource encryption

### PR DESCRIPTION
**Pull request by rmgraham**

### CODE REVIEW

**Review by kizimo — APPROVED** [review]

**Review by keithmorganIBM — DISMISSED** [review]

**Comment by dalehille** [comment]
@rmgraham I'm seeing some errors when running things locally.  This is after I updated my org with `enableResourceEncryption` and then ran the `createOrgEncKey` mutation

```[2021-03-16T16:34:48.736Z] ERROR: razeedash-api/clusters/19702 on dale2021.attlocal.net: error:0409A06E:rsa routines:RSA_padding_add_PKCS1_OAEP_mgf1:data too large for key size (req_id=034501d9-32e7-4065-8f24-3533463d447c)
[2021-03-16T16:34:48.736Z] ERROR: razeedash-api/clusters/19702 on dale2021.attlocal.net: error:0409A06E:rsa routines:RSA_padding_add_PKCS1_OAEP_mgf1:data too large for key size (req_id=034501d9-32e7-4065-8f24-3533463d447c, err.code=ERR_OSSL_RSA_DATA_TOO_LARGE_FOR_KEY_SIZE)
    Error: error:0409A06E:rsa routines:RSA_padding_add_PKCS1_OAEP_mgf1:data too large for key size
        at Object.publicEncrypt (internal/crypto/cipher.js:63:12)
        at rsaEncrypt (/Users/dale/code/iks/opensource/razeedash-api/app/utils/orgs.js:115:29)
        at encryptStrUsingOrgEncKey (/Users/dale/code/iks/opensource/razeedash-api/app/utils/orgs.js:85:14)
        at updateClusterResources (/Users/dale/code/iks/opensource/razeedash-api/app/routes/v2/clusters.js:269:37)
        at runMicrotasks (<anonymous>)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Comment by dalehille** [comment]
Are you also going to update deployableVersion encrypt/decrypt code in this PR?  I'm thinking about `getAndDecryptFile` and  `encryptAndUploadFile` from s3Client.js

File: app/routes/cron/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2019 IBM Corp. All Rights Reserved.
```
Comment (adamkingit) [inline app/routes/cron/index.js:2]:
```suggestion
 * Copyright 2021 IBM Corp. All Rights Reserved.
```

File: app/routes/cron/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ */
+const bunyan = require('bunyan');
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const router = express.Router();
+const { getBunyanConfig } = require('../../utils/bunyan');
+const logger = bunyan.createLogger(getBunyanConfig('razeedash-api/cron/index.js'));
+const { cronRotateEncKeys, migrateResourcesToNewOrgKeysCron } = require('../../utils/orgs');
+
+// /cron/rotateEncKeys
```
Comment (adamkingit) [inline app/routes/cron/index.js:24]: How do we ensure rotation is initiated by us, or an authorized caller?

File: app/apollo/resolvers/organization.js [hunk 3]
```diff
@@ -26,6 +31,81 @@ const organizationResolvers = {
...
+        throw new RazeeValidationError(context.req.t('org id was not found'), context);
+      }
+      const liveKeys = _.filter(org.encKeys||[], (encKey)=>{
+        return !encKey.deleted;
+      });
+      if(liveKeys.length >= 10){
+        throw new BasicRazeeError(context.req.t('This org has too many encryption keys. Remove some before adding any new ones'), context);
+      }
+
+      const encKey = await genKey();
```
Comment (adamkingit) [inline app/apollo/resolvers/organization.js:52]:
```suggestion
      const encKey = genKey();
```
genKey doesn't appear to return a promise

File: app/utils/orgs.js [hunk 4]
```diff
@@ -68,4 +73,329 @@ const decryptOrgData = (orgKey, data) => {
...
+  var key = randBuff.toString('base64');
+  var id = uuid();
+  var creationTime = new Date();
+  var deleted = false;
+  return {
+    id, key, creationTime, deleted,
+  };
+};
+
+var cronRotateEncKeys = async({ db, maxAge=1000*60*60*24*365/2 })=>{
```
Comment (adamkingit) [inline app/utils/orgs.js:125]: How do we ensure that we don't rotate the keys while we are already in the process of rotating the keys

**Comment by adamkingit** [comment]
Most uses of `var` should be `const`, or sometimes `let`

**Pull request closed without merging**